hibernate-core/src/test/java/org/hibernate/test/annotations/onetoone/hh11746/OneToOneCascadeMapsId.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.annotations.onetoone.hh11746;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Jonathan Ihm
+ */
+public class OneToOneCascadeMapsId extends BaseCoreFunctionalTestCase {
+    @Test
+    public void hhh11746Test() throws Exception {
+        // BaseCoreFunctionalTestCase automatically creates the SessionFactory and provides the Session.
+        Session s = openSession();
+        Transaction tx = s.beginTransaction();
+        insertTestData(s);
+        tx.commit();
+
+        s.clear();
+        tx = s.beginTransaction();
+        Person person = s.find(Person.class, new PersonId("a1", "s1"));
+        Address adr = person.getAddress();
+        assertEquals(person.getId(), adr.getId());
+
+        person = s.find(Person.class, new PersonId("a2", "s2"));
+        adr = person.getAddress();
+        assertEquals(person.getId(), adr.getId());
+
+        tx.commit();
+        s.close();
+    }
+
+
+
+    private void insertTestData(Session s) {
+        Person person1 = new Person();
+        person1.setId(new PersonId("p1", "s1"));
+        Address address1 = new Address();
+        address1.street = "address1";
+        person1.setAddress(address1);
+        s.persist(person1);
```

Comment:
The point behind `@MapsId` is that you shouldn't need to assign the value inside `PersonId` for the value mapped to the identifier of `Address`.  Therefore, I think these blocks should likely read as:
<pre>
final Address address1 = new Address();
address1.setId( "a1" );
address1.setStreet( "address1" );
final Person person1 = new Person();
final PersonId personId1 = new PersonId();
personId1.setSocialId( "s2" );
person1.setId( personId1 );
person1.setAddress( address1 );
session.persist( person1 );
</pre>
The key difference here to note is that I am assigning the identifier on `Address` (which I should) and then Hibernate is able to replicate that identifier into `PersonId` when it saves `Person`.

Unless I am mistaken, it seems based on the JIRA comments that the user is attempting to use `@MapsId` in the inverse way, forcing an identifier value from the embeddable `PersonId` to the entity `Address`.